.github/workflows/docker-image.yml

```diff
@@ -4,20 +4,46 @@ on:
   push:
     branches:
       - master
+  release:
+    types: [created]
 
 jobs:
   publish:
     name: Build and publish
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version:
+          - 36
+          - 37
+          - 38
+    env:
+      IMAGE: openmined/tenseal
 
     steps:
       - uses: actions/checkout@v2
-      
-      - uses: docker/build-push-action@v1
-        with:
-          username: ${{ secrets.DOCKER_LOGIN }}
-          password: ${{ secrets.DOCKER_PASSWORD }}
-          repository: openmined/tenseal
-          tags: latest
-          tag_with_sha: true
-          add_git_labels: true
+
+      - name: build image
+        run: docker build -t $IMAGE -f docker-images/Dockerfile-py${{ matrix.python-version }} .
+
+      - name: docker login
+        run: echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u ${{ secrets.DOCKER_LOGIN }} --password-stdin
+
+      - name: tag and push latest nightly images
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```

Comment:
The push event is only on master, why do we need to check github.ref?

Reply:
double checking, sorry I forget that.